applications/StructuralMechanicsApplication/python_scripts/symbolic_generation/total_lagrangian_mixed_volumetric_strain_element/total_lagrangian_mixed_volumetric_strain_element_template.cpp

```diff
@@ -0,0 +1,1261 @@
+    const ConstitutiveLaw::StressMeasure ThisStressMeasure) const
+{
+    // Set the constitutive variables
+    SetConstitutiveVariables(rThisKinematicVariables, rThisConstitutiveVariables, rValues, PointNumber, IntegrationPoints);
+
+    // Actually do the computations in the ConstitutiveLaw
+    mConstitutiveLawVector[PointNumber]->CalculateMaterialResponse(rValues, ThisStressMeasure); //here the calculations are actually done
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void TotalLagrangianMixedVolumetricStrainElement<TDim>::CalculateKinematicVariables(
+    KinematicVariables& rThisKinematicVariables,
+    const IndexType PointNumber,
+    const GeometryType::IntegrationMethod& rIntegrationMethod) const
+{
+    const auto& r_geometry = GetGeometry();
+    const auto& r_integration_points = r_geometry.IntegrationPoints(rIntegrationMethod);
+
+    // Shape functions
+    rThisKinematicVariables.N = r_geometry.ShapeFunctionsValues(rThisKinematicVariables.N, r_integration_points[PointNumber].Coordinates());
+
+    // Calculate the inverse Jacobian
+    GeometryUtils::JacobianOnInitialConfiguration(
+        r_geometry,
+        r_integration_points[PointNumber],
+        rThisKinematicVariables.J0);
+    MathUtils<double>::InvertMatrix(
+        rThisKinematicVariables.J0,
+        rThisKinematicVariables.InvJ0,
+        rThisKinematicVariables.detJ0);
+    KRATOS_ERROR_IF(rThisKinematicVariables.detJ0 < 0.0)
+        << "Element ID: " << this->Id() << " is inverted. det(J0) = " << rThisKinematicVariables.detJ0 << std::endl;
+
+    // Calculate the shape functions gradients
+    GeometryUtils::ShapeFunctionsGradients(
+        r_geometry.ShapeFunctionsLocalGradients(rIntegrationMethod)[PointNumber],
+        rThisKinematicVariables.InvJ0,
+        rThisKinematicVariables.DN_DX);
+
+    // Calculate the equivalent total strain
+    CalculateEquivalentStrain(rThisKinematicVariables);
+
+    // Compute equivalent F
+    CalculateEquivalentF(rThisKinematicVariables);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void TotalLagrangianMixedVolumetricStrainElement<2>::CalculateEquivalentStrain(KinematicVariables& rThisKinematicVariables) const
+{
+    // Define references to the auxiliary symbols
+    const auto& N = rThisKinematicVariables.N;
+    const auto& DN = rThisKinematicVariables.DN_DX;
+    const auto& u = rThisKinematicVariables.Displacements;
+    const auto& th = rThisKinematicVariables.JacobianDeterminant;
+    auto& r_eq_green_strain = rThisKinematicVariables.EquivalentStrain;
+
+    // Fill the equivalent Green strain values
+    //substitute_green_strain_2D_3N
```

Comment:
Idem I would move it to an independent function so it can reduce code duplication

Reply:
See my answer above.